### DISCUSSION TITLE
Dynamically set directoryPath

### DISCUSSION THREAD

**Discussion by NeshaLe** [khaosdoctor/multer-firebase-storage#2] [Q&A]
Hello,
Thanks for a very nice & functional library. I do have one question though: is it possible to set directoryPath dynamically? For example - directoryPath: `${some folder}/${some subfolder}` (in backticks)
Best,
Nesha

**Comment by khaosdoctor** [comment]
Yes definitely, you just need to pass on this property to the `FirebaseStorage` in the options here: https://github.com/khaosdoctor/multer-firebase-storage#optional-options

It just sends the value you pass on to the multer handler and then it should be dynamically set, however, you'd need a different instance for each path, since this is set up on the initialization

However, you can dynamically set those via hooks: https://github.com/khaosdoctor/multer-firebase-storage#lifecycle-hooks

**Reply by NeshaLe** [reply]
Thank you very much for replying. Unfortunately, it didn't work since the parameters I needed were on the request object. Using the beforeUpload hook sounded like a great idea, but it didn't work as well - or in both cases I was doing something wrong. IF it would work it would make the library more valuable since it'd make the code reusable, just dynamically changing firebase directoryPath and/or prefixing or suffixing the name of the uploader (from the user object on the request object, for example).  EDIT: Got it to work as I wanted, thanks.

**Reply by khaosdoctor** [reply]
Can you please share what you did? And what was your problem? This way I can take a better look and see if there's a way to improve the library to make it cover that use case :)
